megamek/src/megamek/common/Board.java

```diff
@@ -971,12 +971,40 @@ private boolean isValid(IHex[] data, int width, int height, StringBuffer errBuff
                 }
                 StringBuffer currBuff = new StringBuffer();
                 boolean valid = hex.isValid(currBuff);
+                
+                // Multi-hex problems 
+                // A building hex must only have exits to other building hexes of the same Building Type and Class
+                if (hex.containsTerrain(Terrains.BUILDING) && hex.getTerrain(Terrains.BUILDING).hasExitsSpecified()) {
+                    for (int dir = 0; dir < 6; dir++) {
+                        IHex adjHex = getHexInDir(x, y, dir);
+                        if (adjHex != null 
+                                && adjHex.containsTerrain(Terrains.BUILDING) 
+                                && hex.containsTerrainExit(Terrains.BUILDING, dir)) 
+                        {
+                            if (adjHex.getTerrain(Terrains.BUILDING).getLevel() != hex.getTerrain(Terrains.BUILDING).getLevel()) {
+                                valid = false;
+                                currBuff.append("Building has an exit to a building of another Building Type (Light, Medium...).\n");
+                            }
+                            if (hex.containsTerrain(Terrains.BLDG_CLASS) 
+                                    && ((adjHex.containsTerrain(Terrains.BLDG_CLASS) 
+                                            && adjHex.getTerrain(Terrains.BLDG_CLASS).getLevel() != hex.getTerrain(Terrains.BLDG_CLASS).getLevel())
+                                            || (!adjHex.containsTerrain(Terrains.BLDG_CLASS) )  )) {
+                                valid = false;
+                                currBuff.append("Building has an exit in direction "+dir+" to a building of another Building Class.\n");
```

Comment:
Replace the + dir + with chained append calls.

Reply:
Why? much longer. Effortlessly readable as is.

Reply:
Faster processing

Reply:
For the sake of not arguing about irrelevant issues I've changed it. I also followed most of your suggestions above (I think I just didnt add the parenthesis clutter in Terrain.java as this would be inconsistent :D but everything else). Thanks for looking at it.